src/utils/react-query-factory.js

```diff
@@ -0,0 +1,56 @@
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+
+export const queryKeys = {
+  // Global key to all queries
+  all: ['enterprise'],
+  enterprise: (enterpriseUuid) => [...queryKeys.all, enterpriseUuid],
+  enterpriseLearner: () => [...queryKeys.all, 'linked-enterprise-customer-users'],
```

Comment:
[question] How does this interact with `enterpriseSlug`?  On initial page load of the root route (`https://localhost.stage.edx.org:8734/`), `enterpriseSlug` is undefined.

Reply:
Running this query with a yet to be defined slug, for instance if the endpoint is `localhost:8734/` we notice that the endpoint key is null, before it is resolved with the correct query key.

<img width="977" alt="Screenshot 2024-02-23 at 12 19 20 PM" src="https://github.com/openedx/frontend-app-learner-portal-enterprise/assets/82611798/5e8f0e03-c2e4-4472-8108-fb04df83b037">

Reply:
On a fresh page load where the slug is specifically defined as `undefined`
We do see the following error on the Layout page, but a subsequent page refresh will load the page. Mitigating this issue now. 

<img width="1508" alt="Screenshot 2024-02-23 at 12 29 55 PM" src="https://github.com/openedx/frontend-app-learner-portal-enterprise/assets/82611798/d4081099-2e68-4e98-acc1-51e4cc25fa76">

Reply:
The following condition should resolve the issue above.
By allowing an object to fallback to an undefined value with the ?, we can enter the `NotFoundPage` fallback. The issue became the page refresh would make the NotFoundPage perceptible for just a second before correctly resolving. So wrapping it in a Delayed fallback with the default imperceptible time corrected the undefined issue.
Layout.jsx
```js
  if (!enterpriseLearnerData?.enterpriseCustomer) {
    return (
      <DelayedFallbackContainer
        className="py-5 text-center"
        screenReaderText="Loading your account details. Please wait."
      >
        <NotFoundPage />;
      </DelayedFallbackContainer>
    );
  }
  ```